GVFS/GVFS.Platform.Windows/WindowsFileSystem.cs

```diff
@@ -1,16 +1,77 @@
 ﻿using GVFS.Common;
 using GVFS.Common.FileSystem;
+using GVFS.Common.Tracing;
 using Microsoft.Win32.SafeHandles;
 using System;
 using System.IO;
 using System.Runtime.InteropServices;
+using System.Security.AccessControl;
+using System.Security.Principal;
 
 namespace GVFS.Platform.Windows
 {
     public partial class WindowsFileSystem : IPlatformFileSystem
     {
         public bool SupportsFileMode { get; } = false;
 
+        /// <summary>
+        /// Adds a new FileSystemAccessRule granting read (and optionally modify) access for all users.
+        /// </summary>
+        /// <param name="directorySecurity">DirectorySecurity to which a FileSystemAccessRule will be added.</param>
+        /// <param name="grantUsersModifyPermissions">
+        /// True if all users should be given modify access, false if users should only be allowed read access
+        /// </param>
+        public static void AddUsersAccessRulesToDirectorySecurity(DirectorySecurity directorySecurity, bool grantUsersModifyPermissions)
+        {
+            SecurityIdentifier allUsers = new SecurityIdentifier(WellKnownSidType.BuiltinUsersSid, null);
+            FileSystemRights rights = FileSystemRights.Read;
+            if (grantUsersModifyPermissions)
+            {
+                rights = rights | FileSystemRights.Modify;
+            }
+
+            directorySecurity.AddAccessRule(
+                new FileSystemAccessRule(
+                    allUsers,
+                    rights,
+                    InheritanceFlags.ContainerInherit | InheritanceFlags.ObjectInherit,
+                    PropagationFlags.None,
+                    AccessControlType.Allow));
+        }
+
+        /// <summary>
+        /// Adds a new FileSystemAccessRule granting read/exceute/modify/delete access for administrators.
+        /// </summary>
+        /// <param name="directorySecurity">DirectorySecurity to which a FileSystemAccessRule will be added.</param>
+        public static void AddAdminAccessRulesToDirectorySecurity(DirectorySecurity directorySecurity)
+        {
+            SecurityIdentifier administratorUsers = new SecurityIdentifier(WellKnownSidType.BuiltinAdministratorsSid, null);
+            directorySecurity.AddAccessRule(

```

Comment:
A comment about why we chose these particular values might be useful - I know we spent a bit of time going through some of the combinations.